pallets/registrar/src/lib.rs

```diff
@@ -995,6 +997,12 @@ pub mod pallet {
                 }
             }
 
+            // Call hook for newly assigned registered chains
+            if let Some(ref new_paras) = new_paras {
+                new_paras.iter().for_each(|para_id| {
+                    T::CollatorAssignmentHook::on_collators_assigned(*para_id);
```

Comment:
This is the list of para ids that want collators to be assigned, before filtering out those that do not have credits. So it is possible for a container chain to get charged and get no collators. A better place for this hook would be in pallet_collator_assignment, after we know which container chains will be assigned collators.

Reply:
perfect thanks @tmpolaczyk